### PR TITLE
Vertoz Adapter

### PR DESCRIPTION
**Pull request by Prebid-Vertoz**
## Type of change

- [x] New bidder adapter
## Description of change

      Added Vertoz adapter

- test parameters for validating bids
```
{
  bidder: 'vertoz',
  params: {
        placementId:'VZH123'      //REQUIRED AD unit id
  }
}
```
- contact email of the adapter’s maintainer : prebid-team@vertoz.com
- [x] official adapter submission

## Other information

@jaiminpanchal27 
- [x] Added bid object as second parameter in createBid call.
- [x] Use of constants defined in constants.json.
- [x] Removed extra lines from vertoz_spec.js.
- [x] Fixed endpoint access to global.

### CODE REVIEW

File: src/adapters/vertoz.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  const QUERY_PARAM_KEY='q';
+
+  function _callBids(params) {
+
+    var bids = params.bids || [];
+
+    for (var i = 0; i < bids.length; i++) {
+
+      var bid = bids[i]; 
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  function _callBids(params) {
+
+    var bids = params.bids || [];
+
+    for (var i = 0; i < bids.length; i++) {
+
+      var bid = bids[i]; 
+    	 	
+      let slotBidId = utils.getValue(bid,'bidId');
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      let cb = Math.round(new Date().getTime() / 1000);
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      let vzEndPoint = BASE_URI;
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      let reqParams = bid.params || {};
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      let placementId = utils.getValue(reqParams,'placementId');
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+      var vzReq = {
+        _vzPlacementId:placementId,
+        _rqsrc:reqSrc,
+        _cb:cb,
+        _slotBidId:slotBidId
+      };
+	
+      let queryParamValue=JSON.stringify(vzReq);
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+        _vzPlacementId:placementId,
+        _rqsrc:reqSrc,
+        _cb:cb,
+        _slotBidId:slotBidId
+      };
+	
+      let queryParamValue=JSON.stringify(vzReq);
+      
+      vzEndPoint = utils.tryAppendQueryString(vzEndPoint, QUERY_PARAM_KEY, queryParamValue);
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+      	
+      adloader.loadScript(vzEndPoint);
+    }
+
+  }
+
+  $$PREBID_GLOBAL$$.vzResponse = function (vertozResponse) {
+
+    let bidRespObj = vertozResponse;
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+  }
+
+  $$PREBID_GLOBAL$$.vzResponse = function (vertozResponse) {
+
+    let bidRespObj = vertozResponse;
+   
+    var bidObject;
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+    var reqBidObj=utils.getBidRequest(bidRespObj.slotBidId);
+    
+    if (bidRespObj.cpm) {
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+      bidObject = bidfactory.createBid(CONSTANTS.STATUS.GOOD,reqBidObj);
+      bidObject.cpm = Number(bidRespObj.cpm);
+      bidObject.ad = bidRespObj.ad + utils.createTrackPixelHtml(decodeURIComponent(bidRespObj.nurl));
+      bidObject.width = bidRespObj.adWidth;
+      bidObject.height = bidRespObj.adHeight;
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+      bidObject = bidfactory.createBid(CONSTANTS.STATUS.GOOD,reqBidObj);
+      bidObject.cpm = Number(bidRespObj.cpm);
+      bidObject.ad = bidRespObj.ad + utils.createTrackPixelHtml(decodeURIComponent(bidRespObj.nurl));
+      bidObject.width = bidRespObj.adWidth;
+      bidObject.height = bidRespObj.adHeight;
+	
+    }  
+    else {
+
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: Remove line

File: src/adapters/vertoz.js [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+    if (bidRespObj.cpm) {
+
+      bidObject = bidfactory.createBid(CONSTANTS.STATUS.GOOD,reqBidObj);
+      bidObject.cpm = Number(bidRespObj.cpm);
+      bidObject.ad = bidRespObj.ad + utils.createTrackPixelHtml(decodeURIComponent(bidRespObj.nurl));
+      bidObject.width = bidRespObj.adWidth;
+      bidObject.height = bidRespObj.adHeight;
+	
+    }  
+    else {
```
Comment (jaiminpanchal27) [inline src/adapters/vertoz.js]: if else should be like, here else is not in proper line.
```
if {
//Code
} else {
//Code
}
```

File: test/spec/adapters/vertoz_spec.js [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+        });
+
+    });
+
+    describe('Bid response', () => {
+
+        let vzBidRequest;
+
+        let bidderReponse = {
+
```
Comment (jaiminpanchal27) [inline test/spec/adapters/vertoz_spec.js]: Remove line

File: test/spec/adapters/vertoz_spec.js [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+        beforeEach(() => {
+            pbjs._bidsRequested.push(bidderRequest);
+        });
+
+        describe('success', () => {
+
+            let firstBidReg;
+            let adSpaceId;
+
```
Comment (jaiminpanchal27) [inline test/spec/adapters/vertoz_spec.js]: Remove line

**Comment by Nilesh5** [comment]
@jaiminpanchal27 
Hello Jaimin,
I understood the changes and have edited accordingly.
Please review the changes.
Thank You.

**Comment by jaiminpanchal27** [comment]
Thanks @Prebid-Vertoz 
Looks good. Merging

**Comment by mkendall07** [comment]
@Prebid-Vertoz 
Please submit a docs PR here: https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders

**Comment by Nilesh5** [comment]
@mkendall07
Yes sure,Matt.
Will do it soon.

**Pull request merged**